command/write.go

```diff
@@ -147,10 +148,83 @@ func (c *WriteCommand) Run(args []string) int {
 	}
 
 	if secret != nil && secret.Auth != nil && secret.Auth.MFARequirement != nil {
+		if len(secret.Auth.MFARequirement.MFAConstraints) == 1 && !c.flagNonInteractive {
+			// Currently, if there is only one MFA method configured, the login
+			// request is validated interactively
+			methodID, usePasscode := c.getMethodIDUsePasscode(secret.Auth.MFARequirement.MFAConstraints)
+			if methodID != "" {
+				return c.validateMFA(secret.Auth.MFARequirement.MFARequestID, methodID, usePasscode)
+			}
+		}
 		c.UI.Warn(wrapAtLength("A login request was issued that is subject to "+
 			"MFA validation. Please make sure to validate the login by sending another "+
 			"request to mfa/validate endpoint.") + "\n")
 	}
+
+	// Handle single field output
+	if c.flagField != "" {
+		return PrintRawField(c.UI, secret, c.flagField)
+	}
+
+	return OutputSecret(c.UI, secret)
+}
+
+func (c *WriteCommand) getMethodIDUsePasscode(mfaConstraintAny map[string]*logical.MFAConstraintAny) (string, bool) {
+	for _, mfaConstraint := range mfaConstraintAny {
+		if len(mfaConstraint.Any) != 1 {
+			return "", false
+		}
+		return mfaConstraint.Any[0].ID, mfaConstraint.Any[0].UsesPasscode
+	}
+	return "", false
+}
+
+func (c *WriteCommand) validateMFA(reqID, mfaMethodID string, usePasscode bool) int {
+	var passcode string
+	var err error
+	if usePasscode {
+		passcode, err = c.UI.AskSecret(fmt.Sprintf("Enter the passphrase for methodID %q:", mfaMethodID))
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("failed to read the passphrase with error %q. please validate the login by sending a request to mfa/validate", err.Error()))
+			return 2
+		}
+	} else {
+		// TODO: not sure about printing this message; an error might occur before or during the validate request
+		c.UI.Warn("Please acknowledge the push notification in your authenticator app")
```

Comment:
Shouldn't we prompt them (e.g. "hit enter") so we know when they've done it?

Reply:
Well, at this point, the validate request has not issued yet. This is just a reminder for them to check their authenticator app. There is a chance that the validate request fails even before issuing the notification.

Reply:
I see.  Maybe we can make the message a little more explicit then, something like "Asking Vault to perform MFA validation with upstream service.  You should receive a push notification in your authenticator app shortly"?  Otherwise if they go and look immediately they might not see the push yet.